src/main/java/org/sonarsource/scanner/jenkins/pipeline/WaitForQualityGateStep.java

```diff
@@ -196,7 +227,7 @@ public void onTaskCompleted(String taskId, String taskStatus, @Nullable String q
         SonarQubeWebHook.get().removeListener(this);
         switch (taskStatus) {
           case WsClient.CETask.STATUS_SUCCESS:
-            getContext().onSuccess(new QGStatus(qgStatus));
+            handleQGStatus(qgStatus);
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Annotate the parameter with @javax.annotation.Nullable in method 'handleQGStatus' declaration, or make sure that null can not be passed as argument. ([squid:S4449](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS4449))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.jenkins-ci.plugins%3Asonar&pullRequest=90&issues=AWI9oBgHdpSr17abUG11&open=AWI9oBgHdpSr17abUG11)

[ISSUE_KEY]: AWI9oBgHdpSr17abUG11